### PR TITLE
port `gridss/gridss` to nf-test

### LINKED ISSUE #5949: `--assembly` flag of gridss module leads to empty bam files
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

GRIDSS allows to set a flag called `--assembly` which should report a `bam` file as a part of the structural variant calling algorithm that is implemented in the tool. This was also added to the nf-core module. While porting to nf-test I realised that the created files were always empty and the exact way to set the flag was unclear so I asked the tool developers: https://github.com/PapenfussLab/gridss/issues/664. For now I suggest removing the flag from the module and add it later when the usage is better documented.

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #5949 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by famosab** [comment]
@nf-core-bot fix linting please

File: modules/nf-core/gridss/gridss/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            process {
+                """
+                input[0] = [ [ id:'test' ], // meta map
+                        file(params.modules_testdata_base_path + '/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true), //inputs
+                        [] //assembly
+                        ]
+                input[1] = [ [id:'fasta'],
+                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
+                        ]
+                input[2] = [ [id:'fasta_fai'],
```
Comment (SPPearce) [inline modules/nf-core/gridss/gridss/tests/main.nf.test:40]: Indentation is a bit funky here, especially compared to input[1]

File: modules/nf-core/gridss/gridss/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+                            file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
```
Comment (SPPearce) [inline modules/nf-core/gridss/gridss/tests/main.nf.test:71]: And indentation here

File: modules/nf-core/gridss/gridss/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                input[3] = [ [], [] ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    file(process.out.assembly[0][1]).name,
```
Comment (SPPearce) [inline modules/nf-core/gridss/gridss/tests/main.nf.test:112]: Is this a bam file? As there is a [nf-test extension for bam files](https://nvnieuwk.github.io/nft-bam/latest/)

Comment (famosab) [reply modules/nf-core/gridss/gridss/tests/main.nf.test:112]: Is there another module who uses this were I can have a look?

File: modules/nf-core/gridss/gridss/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
```
Comment (nvnieuwk) [inline modules/nf-core/gridss/gridss/tests/main.nf.test:112]:
```suggestion
                    bam(process.out.assembly[0][1]).getSamLinesMD5(),
```

This should do it! You can have a look at the [plugin documentation](https://nvnieuwk.github.io/nft-bam/latest/usage/) for examples :)

Comment (famosab) [reply modules/nf-core/gridss/gridss/tests/main.nf.test:112]: Nice! Just FYI: the link to the examples at the bottom of the docs is broken.

Comment (nvnieuwk) [reply modules/nf-core/gridss/gridss/tests/main.nf.test]: Should be fixed now! Thanks for reporting :)

**Comment by famosab** [comment]
@nvnieuwk It seems to me that the bam is either not created or always empty. I cannot seem to figure out why that is. I thought I detected a bug in the script part, but now I am unsure. I think my problem is that I do not understand what gridss actually wants as input for `--assembly`. Maybe you can have a look 👍

**Comment by nvnieuwk** [comment]
I have no idea myself what the `--assembly` option is for :/ I don't use it myself

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**